plugin/environment.go

```diff
@@ -300,10 +301,14 @@ func (env *Environment) Shutdown() {
 		ap := value.(activePlugin)
 
 		if ap.supervisor != nil {
-			if err := ap.supervisor.Hooks().OnDeactivate(); err != nil {
-				env.logger.Error("Plugin OnDeactivate() error", mlog.String("plugin_id", ap.BundleInfo.Manifest.Id), mlog.Err(err))
+			//if err := ap.supervisor.Hooks().OnDeactivate(); err != nil {
+			//	env.logger.Error("Plugin OnDeactivate() error", mlog.String("plugin_id", ap.BundleInfo.Manifest.Id), mlog.Err(err))
+			//}
+			//ap.supervisor.Shutdown()
+			p, err := os.FindProcess(ap.supervisor.pid)
```

Comment:
I don't understand the exact role of the supervisor, but as a general rule we should shut things down gracefully. 1/5 The supervisor then should attempt to kill the plugins, however it decides fit (including just exiting if it works).

Reply:
I would prefer it in too.   however, the `ap.supervisor.Shutdown` process calls `client.Kill()`, which is the hashicorp go-plugin client code, and seems to conflict with the direct `p.Signal` call.   (see above on the combined issue canceling out the pid signal)

Reply:
@levb I can look further into the hashicorp go-plugin code (perhaps we can find a bug there), though we are on their latest release https://github.com/hashicorp/go-plugin/releases